.jenkins_nightly

```diff
@@ -0,0 +1,77 @@
+pipeline {
+    agent none
+
+    options {
+        timeout(time: 6, unit: 'HOURS')
+    }
+
+    stages {
+        stage('Build') {
+            parallel {
+                stage('spack-serial') {
+                    agent {
+                        docker {
+                          image 'ubuntu:22.04'
+                          label 'docker'
+                        }
+                    }
+                    steps {
+                        sh '''
+                          DEBIAN_FRONTEND=noninteractive && \
+                          apt-get update && apt-get upgrade -y && apt-get install -y \
+                          build-essential \
+                          wget \
+                          git \
+                          bc \
+                          python3-dev \
+                          && \
+                          apt-get clean && rm -rf /var/lib/apt/lists/*
+
+                          rm -rf spack && \
+                          git clone https://github.com/spack/spack.git && \
+                          . ./spack/share/spack/setup-env.sh && \
+                          spack install kokkos@develop+tests && \
+                          spack load cmake && \
+                          spack test run kokkos && \
+                          spack test results -l
+                          '''
+                    }      
+                }   
+                stage('spack-cuda') {
+                    agent {
+                        docker {
+                          image 'nvidia/cuda:12.1.0-devel-ubuntu22.04'
+                          label 'nvidia-docker && ampere'
+                        }
+                    }
+                    steps {
+                        sh '''
+                          DEBIAN_FRONTEND=noninteractive && \
+                          apt-get update && apt-get upgrade -y && apt-get install -y \
+                          build-essential \
+                          wget \
+                          git \
+                          bc \
+                          python3-dev \
+                          gfortran \
```

Comment:
where is gfortran used?  some of our examples?

Reply:
We need gfortran for openmpi... In spack, `nvcc_wrapper` depends on MPI because we set `OMPI_CXX`, etc. Once I understand better how spack sets variables, I'll remove the MPI dependency.